spatialdata/_core/_spatialdata.py

```diff
@@ -1,59 +1,83 @@
-import json
-
-# import colorama
-# colorama.init(strip=False)
-import tempfile
+import hashlib
 import os
-import filecmp
+import tempfile
+from types import MappingProxyType
+from typing import Any, List, Mapping, Optional, Set, Tuple
 
 import numpy as np
 import xarray as xr
-import hashlib
-from types import MappingProxyType
-from typing import Any, Mapping, Optional
-
+import zarr
 from anndata import AnnData
-from anndata._io import read_zarr
 from ome_zarr.io import parse_url
-from ome_zarr.reader import Reader
 
 from spatialdata._core.mixin.io_mixin import IoMixin
-from spatialdata._core.writer import write_spatial_anndata
-from spatialdata._core.transform import Transform, set_transform, get_transform
+
+# from spatialdata._core.writer import write_spatial_anndata
+from spatialdata._core.transform import Transform, get_transform, set_transform
+from spatialdata._core.writer import (
+    write_image,
+    write_labels,
+    write_points,
+    write_shapes,
+    write_tables,
+)
 from spatialdata.utils import are_directories_identical
 
 
 class SpatialData(IoMixin):
     """Spatial data structure."""
 
-    adata: AnnData
-    regions: Mapping[str, Any]
-    images: Mapping[str, Any]
-    points: Optional[AnnData]
+    tables: Optional[AnnData]
+    labels: Optional[Mapping[str, Any]]
+    images: Optional[Mapping[str, Any]]
+    points: Optional[Mapping[str, AnnData]]
+    shapes: Optional[Mapping[str, AnnData]]
+    elems: List[Optional[Set[str]]]
 
     def __init__(
         self,
-        adata: Optional[AnnData] = None,
-        regions: Mapping[str, Any] = MappingProxyType({}),
-        images: Mapping[str, Any] = MappingProxyType({}),
+        tables: Optional[AnnData] = None,
+        labels: Optional[Mapping[str, Any]] = MappingProxyType({}),
```

Comment:
To decide if we will add polygons and "star regions" as 2 (+ 2 because of the transforms) different arguments, or if we just allow for a mapping of regions and a mapping of transforms

Reply:
to be discussed, let's keep it for now